subsys/fs/nvs/nvs.c

```diff
@@ -17,6 +17,26 @@
 LOG_MODULE_REGISTER(fs_nvs, CONFIG_NVS_LOG_LEVEL);
 
 /* basic routines */
+
+/* Get the base address of the sector where the address points into */
+static inline uint32_t sector_base_addr(struct nvs_fs *fs, uint32_t addr)
+{
+	return fs->sector_size * (addr >> NVS_OFFSET_BITS);
+}
+
+/* Get the offset from the addresses to the next lower sector start */
+#if NVS_OFFSET_BITS > 16
```

Comment:
Can you put the #if instead the function? What is the point of using uint16_t as a return value? May we wlll just use unsigned int?